### PR TITLE
BMC Remedy Integration Pack

### PR DESCRIPTION
**Pull request by renancaldeira**
This is a first version for BMC Remedy Integration Pack

### CODE REVIEW

**Comment by jfryman** [comment]
Very nice!

I see a few linting issues (super small... like # of spaces between imports and class declaration in a few of the actions). but those were not exposed to you automatically because I disabled the Travis CI check on PR. Sorry about that!

Anyway, instead of enumerating through them, would you either...

* Run the tests locally `make` and see what pops up, or...
* Push another commit to re-trigger Travis

I don't see a re-trigger option with the new GitHub -> Travis integration, so apologies on asking you to do this. However, that should at least get those squared away that we can merge this in.

The rest looks nice! I especially like the sensor contribution! :heart_eyes: 

Thanks so much for the contribution! :bow:

File: packs/remedy/sensors/actions/delete.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from lib.actions import BaseAction
```
Comment (dzimine) [inline packs/remedy/sensors/actions/delete.py]: This and following actions are under `packs/remedy/sensors`, is this omission, or intention?

File: packs/remedy/sensors/workorder_sensor.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+from st2reactor.sensor import PollingSensor
+from actions.lib.actions import BaseAction
+from actions import *
+
+class SimplePollingSensor(PollingSensor):
```
Comment (dzimine) [inline packs/remedy/sensors/workorder_sensor.py]: I suggest to name the class by the sensor name: `WorkorderSensor`

File: packs/remedy/sensors/workorder_sensor.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        #   # E.g.: dispatch('examples.foo_sensor', {'k1': 'stuff', 'k2': 'foo'})
+        #   # trace_tag is a tag you would like to associate with the dispacthed TriggerInstance
+        #   # Typically the trace_tag is unique and a reference to an external event.
+        entries = self.client.query(schema='TESTE_RENAN', qualifier='1=1', 
+                                    fields=['Request ID', 'Short Description', 'Status'], 0, 5)
+        
+        for entry_id, entry_values in entries:
+            
+            #self.client.update()
+            print('Entry ID: {}'.format(entry_id))
```
Comment (dzimine) [inline packs/remedy/sensors/workorder_sensor.py]: use log instead, https://docs.stackstorm.com/sensors.html#get-logger-name, example:
```
self.log = self._sensor_service.get_logger(__name__) # in setup
...
self.log.debug("....")
```

File: packs/remedy/sensors/workorder_sensor.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        #   # registered for the trigger.
+        #   self._sensor_service.dispatch(trigger, payload)
+        #   # You can refer to the trigger as dict
+        #   # { "name": ${trigger_name}, "pack": ${trigger_pack} }
+        #   # or just simply by reference as string.
+        #   # i.e. dispatch(${trigger_pack}.${trigger_name}, payload)
+        #   # E.g.: dispatch('examples.foo_sensor', {'k1': 'stuff', 'k2': 'foo'})
+        #   # trace_tag is a tag you would like to associate with the dispacthed TriggerInstance
+        #   # Typically the trace_tag is unique and a reference to an external event.
+        entries = self.client.query(schema='TESTE_RENAN', qualifier='1=1',
```
Comment (dzimine) [inline packs/remedy/sensors/workorder_sensor.py]: I miss where self.client is initialized (granted didn't try to run it). 

BTW we just posted tools to unit-test actions and sensors, including good mocks, they may come handy.
docs: https://docs.stackstorm.com/development/pack_testing.html
example: https://github.com/StackStorm/st2contrib/blob/master/packs/docker/tests/test_sensor_docker_sensor.py

File: packs/remedy/sensors/workorder_sensor.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
+from st2reactor.sensor import PollingSensor
+from actions.lib.actions import BaseAction
+from actions import *
```
Comment (dzimine) [inline packs/remedy/sensors/workorder_sensor.py]: Wildcard imports are discouraged by PEP8. Why is this needed anyways?

**Comment by Kami** [comment]
https://stackstorm.slack.com/archives/stackstorm/p1449719068011672

File: packs/remedy/sensors/workorder_sensor.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    -
+      name: "event"
+      description: "Work Order Trigger"
+      payload_schema:
+        type: "object"
+        properties:
+          executed_at:
+            type: "string"
+            format: "date-time"
+            default: "2014-07-30 05:04:24.578325"
```
Comment (dzimine) [inline packs/remedy/sensors/workorder_sensor.yaml]: interesting why such default :)

**Comment by renancaldeira** [comment]
Sorry about the errors. I'll recheck entire pack.

File: packs/remedy/sensors/workorder_sensor.py (hunk 2 above)
Comment (manasdk) [inline packs/remedy/sensors/workorder_sensor.py]: Needs to match the value in https://github.com/StackStorm/st2contrib/pull/333/files#diff-9ffe7287116d697cc2b8d56b8217cceeR2.

Therefore use `WorkOrderSensor`

File: packs/remedy/sensors/actions/get.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
+from lib.actions import BaseAction
+
```
Comment (manasdk) [inline packs/remedy/sensors/actions/get.py]: 2 empty lines so pylint is happy

File: packs/remedy/sensors/workorder_sensor.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        - indicates the interval between two successive poll() calls.
+    """
+
+    def setup(self):
+        # Setup stuff goes here. For example, you might establish connections
+        # to external system once and reuse it. This is called only once by the system.
+        pass
+
+    def poll(self):
+        # This is where the crux of the sensor work goes.
```
Comment (manasdk) [inline packs/remedy/sensors/workorder_sensor.py]: Can clean up the commented section since you do not need it any longer.

File: packs/remedy/sensors/workorder_sensor.py [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        
+        for entry_id, entry_values in entries:
+            
+            #self.client.update()
+            print('Entry ID: {}'.format(entry_id))
+            print('-' * 80)
+            for field, value in entry_values.items():
+                print('{}: {}'.format(field, value))
+        print()
+
```
Comment (manasdk) [inline packs/remedy/sensors/workorder_sensor.py]: At the end of this method you likely want to dispatch a few trigger. Seems like this sensor is still WIP.

File: packs/remedy/sensors/workorder_sensor.py [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+        # This method is called when trigger is created
+        pass
+
+    def update_trigger(self, trigger):
+        # This method is called when trigger is updated
+        pass
+
+    def remove_trigger(self, trigger):
+        # This method is called when trigger is deleted
+        pass
```
Comment (manasdk) [inline packs/remedy/sensors/workorder_sensor.py]: Please add a new line to end the file.

File: packs/remedy/sensors/workorder_sensor.yaml (hunk 6 above)
Comment (manasdk) [inline packs/remedy/sensors/workorder_sensor.yaml]: Likely copy-pasta. Feel free to remove default.

**Comment by lakshmi-kannan** [comment]
@renancaldeira Would you mind addressing the comments and taking the PR to completion? Seems like this could be useful to people :).

**Comment by LindsayHill** [comment]
Closing this in favour of #604
